packages/vite/src/node/server/environment.ts

```diff
@@ -162,6 +165,20 @@ export class DevEnvironment extends BaseEnvironment {
     )
   }
 
+  // TODO: when restart is called, close is not called for the previous environment
+  //       that means depOptimizer is initialized but not closed
```

Comment:
Would this be a problem? If we move the depOptimizer initialization to `init`, this won't be a problem. Is that safe to do?

Reply:
Maybe we should call `await this.close()` then extract things from the constructor and also call `init()`. There are some things missing to be re-inited. `_crawlEndFinder` should be recreated, `moduleGraph` too. I don't know if we should reuse things. We don't have any long term resource to keep.